### PR TITLE
swap seach popular in rainbow

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes BX-####
Figma link (if any):

## Screen recordings / screenshots

https://github.com/user-attachments/assets/766c7d48-7791-4071-8fd9-c1e4d805b035

## What to test

- 'popular in rainbow' correctly shows in the swap token selection (with a limit of three results)
- other sections (verified, favorites, popular) don't have repeated tokens, like if usdc is in favorites it should not show up in verified section

### CODE REVIEW

File: src/core/resources/search/tokenSearch.ts [hunk 1]
```diff
@@ -90,43 +86,14 @@ async function tokenSearchQueryFunction({
...
-          `${ETH_ADDRESS}_${ChainId.optimism}`,
-          `${ETH_ADDRESS}_${ChainId.arbitrum}`,
-          `${BNB_BSC_ADDRESS}_${ChainId.bsc}`,
-          `${MATIC_POLYGON_ADDRESS}_${ChainId.polygon}`,
-          `${ETH_ADDRESS}_${ChainId.base}`,
-          `${ETH_ADDRESS}_${ChainId.zora}`,
-          `${ETH_ADDRESS}_${ChainId.avalanche}`,
-          `${ETH_ADDRESS}_${ChainId.blast}`,
-          `${ETH_ADDRESS}_${ChainId.degen}`,
-        ].includes(`${a.uniqueId}_${chainId}`),
```
Comment (greg-schrammel) [inline src/core/resources/search/tokenSearch.ts:120]: replaced this with `isNativeAsset` from `~/core/utils/chains`, fixes some chains like degen that was not getting the right native

**Review by estebanmino — APPROVED** [review]
lgtm

File: src/entries/popup/hooks/useSearchCurrencyLists.ts [hunk 2]
```diff
@@ -508,6 +512,10 @@ export function useSearchCurrencyLists({
       return sections;
     }
 
+    if (popularAssets) {
```
Comment (benisgold) [inline src/entries/popup/hooks/useSearchCurrencyLists.ts:515]: i think you need to make sure there is no overlap between assets in the popular section and assets in other sections. here's the app logic https://github.com/rainbow-me/rainbow/blob/develop/src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

lmk if you want me to explain more

**Review by benisgold — CHANGES_REQUESTED** [review]

File: src/core/state/favorites/index.ts [hunk 3]
```diff
@@ -17,8 +17,8 @@ import {
   ETH_BLAST_ADDRESS,
   ETH_OPTIMISM_ADDRESS,
   ETH_ZORA_ADDRESS,
-  POL_POLYGON_ADDRESS,
   OP_ADDRESS,
+  POL_POLYGON_ADDRESS,
```
Comment (greg-schrammel) [inline src/core/state/favorites/index.ts:21]: eslint sort imports rule

File: src/entries/popup/hooks/useSearchCurrencyLists.ts [hunk 4]
```diff
@@ -66,6 +68,21 @@ const filterBridgeAsset = ({
...
+  assets: SearchAsset[],
+  others: (SearchAsset | undefined | null)[],
+) {
+  const _others = others.filter(Boolean);
+  return assets.filter((asset) => {
+    return !_others.some((other) =>
+      isLowerCaseMatch(other.uniqueId, asset.uniqueId),
+    );
+  });
+}
```
Comment (greg-schrammel) [inline src/entries/popup/hooks/useSearchCurrencyLists.ts:84]: made a more generic function to filter the sections, took the name from [new Set().difference()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/difference)

this changes for example 
```ts
filterAssetsFromBridgeAndAssetToSell(sectionAssets)
``` 
for 
```ts
difference(sectionAssets, [bridgeAsset, assetToSell])
```

I think its more extensible and easier to understand the intention

**Review by DanielSinclair — APPROVED** [review]
Functionality looks good to me

**Pull request merged**